src/Modal.js

```diff
@@ -41,7 +41,7 @@ module.exports = {
         var closeDialog = function() {
             ReactDOM.unmountComponentAtNode(self.getOrCreateContainer());
 
-            if (props && props.onFinished) props.onFinished.apply(null, arguments);
+            if (props && props.onFinished) props.onFinished.call(null, false);
```

Comment:
Looks like this has been the case every since Modal.js was first written in https://github.com/vector-im/vector-web/commit/1b4358624fd0dcd932325bc99f2cb9b616c1a925

Either way, your fix always passes false to onFinished which means the user won't leave then room even if they click OK. The correct fix here is to explicitly call `closeDialog(false)` in the background div onClick handler rather than using it as the onClicm handler directly (since the event handler will be passed an event as an argument which, when it propagates down to the `onFinished` handler, will evaluate to `true`.